### PR TITLE
fix(build): remove EOL distros, cleanup alpine and crossbuild aws2

### PR DESCRIPTION
**Pull request by fffonion**
Please review each commits individually.

### Summary

- Remove alpine related code due to Kong 3.3 EOL. The commit should not be backported to 2.8 branch.
- Remove RHEL7 and Debian10 (planned June 2024).
- Cross build AWS2 x86_64.
- As RHEL7 is removed and AWS2 is cross built, we can now safely bump `actions/*@v3` to `v4` that uses node 20.

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-4549

### CODE REVIEW

**Comment by fffonion** [comment]
good run https://github.com/Kong/kong/actions/runs/9172531818/job/25219517740?pr=13052

**Comment by AndyZhang0707** [comment]
Let's wait for the final decision on the OS support policy from product team before we merge this PR.

**Review by Water-Melon — APPROVED** [review]

**Review by windmgc — APPROVED** [review]

**Pull request merged**
